File: src/main/java/com/bdev/hogwarts_api/service/cabinet/CabinetService.java
package com.bdev.hogwarts_api.service.cabinet;

import com.bdev.hogwarts_api.data.dto.cabinet.Cabinet;

import java.util.List;
import java.util.Optional;

public interface CabinetService {
    boolean exists(long id);
    List<Cabinet> getAllCabinets();
    Optional<Cabinet> getCabinet(long id);
    long saveCabinet(Cabinet cabinet);
    void editCabinet(Cabinet cabinet);
    void deleteCabinet(long id);
}

File: src/main/java/com/bdev/hogwarts_api/service/cabinet/CabinetService.kt
package com.bdev.hogwarts_api.service.cabinet

import com.bdev.hogwarts_api.data.dto.cabinet.ExistingCabinet
import com.bdev.hogwarts_api.data.dto.cabinet.NewCabinet

interface CabinetService {
    fun getAllCabinets(): List<ExistingCabinet>
    fun exists(id: Long): Boolean
    fun getCabinet(id: Long): ExistingCabinet?
    fun saveCabinet(cabinet: NewCabinet): Long
    fun editCabinet(cabinet: ExistingCabinet)
    fun deleteCabinet(id: Long)
}
